roles/openshift_hosted/tasks/storage/glusterfs.yml

```diff
@@ -25,6 +25,18 @@
   set_fact:
     openshift_hosted_registry_fsgroup: "{{ registry_pods.results.results[0]['items'][0].spec.securityContext.fsGroup }}"
 
+- name: Get registry PersistentVolume
+  oc_obj:
+    namespace: "{{ openshift_hosted_registry_namespace }}"
+    state: list
+    kind: pv
+    name: "{{ openshift_hosted_registry_storage_volume_name }}-glusterfs-volume"
+  register: registry_pv
```

Comment:
Add a when condition here:
"openshift_hosted_registry_storage_glusterfs_path_discover"
Put default of false in this role's defaults.

Change register variable name to "l_registry_storage_glusterfs_path_discover"